src/v/cluster/members_backend.cc

```diff
@@ -885,13 +885,12 @@ ss::future<std::error_code> members_backend::reconcile() {
     co_return errc::update_in_progress;
 }
 
-ss::future<std::error_code> members_backend::do_remove_node(
-  model::node_id id, model::offset update_offset) {
+ss::future<std::error_code> members_backend::do_remove_node(model::node_id id) {
     if (_features.local().is_active(
           features::feature::membership_change_controller_cmds)) {
         return _members_frontend.local().remove_node(id);
     }
-    return _raft0->remove_member(id, model::revision_id(update_offset));
+    return _raft0->remove_member(id, model::revision_id{0});
```

Comment:
nit: the fact that this statement is not under conditional makes it look like it is on the happy path (while it is actually the opposite)